diff --git CMakeLists.txt CMakeLists.txt
index 74425a2..35cb8cc 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -110,6 +110,14 @@ install(TARGETS fbjni EXPORT fbjniLibraryConfig
 install(EXPORT fbjniLibraryConfig DESTINATION share/cmake/fbjni
   FILE fbjniLibraryConfig.cmake)
 
+install(EXPORT fbjniLibraryConfig DESTINATION share/cmake/fbjni
+        FILE fbjniConfig.cmake
+        NAMESPACE fbjni::)
+
+install(DIRECTORY $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cxx/> DESTINATION include
+  FILES_MATCHING PATTERN "*.h"
+  PATTERN "")
+
 if(MSVC)
     install(FILES $<TARGET_PDB_FILE:fbjni> DESTINATION ${CMAKE_INSTALL_LIBDIR} OPTIONAL)
     install(TARGETS fbjni DESTINATION ${CMAKE_INSTALL_LIBDIR})
