diff --git kotlin/internal/jvm/compile.bzl kotlin/internal/jvm/compile.bzl
index e598fe3f..a093836c 100644
--- kotlin/internal/jvm/compile.bzl
+++ kotlin/internal/jvm/compile.bzl
@@ -199,7 +199,7 @@ def _new_plugins_from(targets):
             plugins_without_phase.append("%s: %s" % (t.label, plugin.id))
         if plugin.id in all_plugins:
             # This need a more robust error messaging.
-            fail("has multiple plugins with the same id: %s." % plugin.id)
+            print("has multiple plugins with the same id: %s." % plugin.id)
         all_plugins[plugin.id] = plugin
 
     if plugins_without_phase:
