diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index 323de6f..24fa0fc 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -46176,6 +46176,9 @@ var ts;
             getPromisedTypeOfPromise: getPromisedTypeOfPromise,
             getAwaitedType: function (type) { return getAwaitedType(type); },
             getReturnTypeOfSignature: getReturnTypeOfSignature,
+            getTypeOfNode: getTypeOfNode,
+            getLiteralTypeFromPropertyName: getLiteralTypeFromPropertyName,
+            getReducedType: getReducedType,
             isNullableType: isNullableType,
             getNullableType: getNullableType,
             getNonNullableType: getNonNullableType,
@@ -64087,13 +64090,13 @@ var ts;
         // has expanded into `[A<NonNullable<NonNullable<NonNullable<NonNullable<NonNullable<T>>>>>>]`
         // in such cases we need to terminate the expansion, and we do so here.
         function isDeeplyNestedType(type, stack, depth) {
-            if (depth >= 5) {
+            if (depth >= 500) {
                 var identity_1 = getRecursionIdentity(type);
                 var count = 0;
                 for (var i = 0; i < depth; i++) {
                     if (getRecursionIdentity(stack[i]) === identity_1) {
                         count++;
-                        if (count >= 5) {
+                        if (count >= 500) {
                             return true;
                         }
                     }
